### PR TITLE
transformer qwen3 Porting to optimum-habana

### PR DESCRIPTION
**Pull request by rkumar2patel**
This PR ports qwen3 model testing to the optimum-habana library for G2 hardware acceleration. The testing suite provides comprehensive coverage for the qwen3 model implementation on Habana Processing Units (HPUs).

Key changes:

Added complete test suite for qwen3 model with HPU-specific adaptations
Configured HPU device targeting and Habana-specific transformers integration

### CODE REVIEW

File: tests/transformers/tests/models/qwen3/test_modeling_qwen3.py [hunk 1]
```diff
@@ -0,0 +1,731 @@
...
+from ...causal_lm_tester import CausalLMModelTest, CausalLMModelTester
+
+
+class Qwen3ModelTester(CausalLMModelTester):
+    config_class = Qwen3Config
+    if is_torch_available():
+        base_model_class = Qwen3Model
+        causal_lm_class = Qwen3ForCausalLM
+        sequence_class = Qwen3ForSequenceClassification
+        token_class = Qwen3ForTokenClassification
```
Comment (gplutop7) [inline tests/transformers/tests/models/qwen3/test_modeling_qwen3.py]: sequence_classification_class = Qwen3ForSequenceClassification
token_classification_class    = Qwen3ForTokenClassification

File: tests/transformers/tests/causal_lm_tester.py [hunk 2]
```diff
@@ -0,0 +1,454 @@
...
+        return self.config_class(**kwargs)
+
+    def create_and_check_model(
+        self, config, input_ids, token_type_ids, input_mask, sequence_labels, token_labels, choice_labels
+    ):
+        model = self.base_model_class(config=config)
+        model.to(torch_device)
+        model.eval()
+        result = model(input_ids, attention_mask=input_mask)
+        result = model(input_ids)
```
Comment (gplutop7) [inline tests/transformers/tests/causal_lm_tester.py]: The second call overwrites result, so you never verify the masked path; it also doubles compute.

File: tests/transformers/tests/causal_lm_tester.py [hunk 3]
```diff
@@ -0,0 +1,454 @@
...
+        self.mamba_d_conv = mamba_d_conv
+        self.mamba_expand = mamba_expand
+        self.mamba_chunk_size = mamba_chunk_size
+
+    def prepare_config_and_inputs(self):
+        input_ids = ids_tensor([self.batch_size, self.seq_length], self.vocab_size)
+
+        input_mask = None
+        if self.use_input_mask:
+            input_mask = torch.tril(torch.ones_like(input_ids).to(torch_device))
```
Comment (gplutop7) [inline tests/transformers/tests/causal_lm_tester.py]: Did you intentionally make attention_mask lower-triangular? In HF, attention_mask is usually shape [batch, seq] with 1 for real tokens and 0 for padding.

File: tests/transformers/tests/models/qwen3/__init__.py
Comment (gplutop7) [inline tests/transformers/tests/models/qwen3/__init__.py:1]: Suggestion - Add the guards: 

```
if not is_habana_available():
    pytest.skip("HPU not available", allow_module_level=True)

if os.environ.get("HF_HUB_OFFLINE") == "1":
    pytest.skip("Requires HF Hub access (HF_HUB_OFFLINE=1).", allow_module_level=True)
```

Comment (rkumar2patel) [reply tests/transformers/tests/models/qwen3/__init__.py:1]: Since this guard is not present in other optimum-habana models, I have followed the same approach for consistency. If we determine it is required, we should incorporate it across all models rather than selectively, to avoid any inconsistency.
